File: frontend/services/ai/adapters/OpenAIAdapter.ts
/**
 * OpenAI适配器实现
 * 支持文本生成、图片生成、音频处理
 */

import { BaseAdapter } from '../base/BaseAdapter';
import {
  AIProvider,
  TextGenerationRequest,
  TextGenerationResponse,
  TextGenerationChunk,
  ImageGenerationRequest,
  ImageGenerationResponse,
  TextToSpeechRequest,
  SpeechToTextRequest,
  AudioResponse,
  VideoGenerationRequest,
  VideoGenerationResponse,
} from '../types';
import { AIConfigManager } from '../config';

/**
 * OpenAI适配器
 */
export class OpenAIAdapter extends BaseAdapter {
  private readonly defaultBaseUrl = 'https://api.openai.com/v1';

  constructor(apiKey?: string) {
    super(apiKey);
    this.baseUrl = this.defaultBaseUrl;
  }

  getProviderType(): AIProvider {
    return 'openai';
  }

  supportsTextGeneration(): boolean {
    return true;
  }

  supportsImageGeneration(): boolean {
    return true;
  }

  supportsTextToSpeech(): boolean {
    return true;
  }

  supportsSpeechToText(): boolean {
    return true;
  }

  supportsVideoGeneration(): boolean {
    return false; // OpenAI暂不支持视频生成
  }

  getSupportedModels(capability: 'text' | 'image' | 'audio' | 'video'): string[] {
    const models: Record<string, string[]> = {
      text: [
        'gpt-4',
        'gpt-4-turbo',
        'gpt-3.5-turbo',
        'gpt-4o',
      ],
      image: [
        'dall-e-3',
        'dall-e-2',
      ],
      audio: [
        'tts-1',
        'tts-1-hd',
        'whisper-1',
      ],
      video: [],
    };
    return models[capability] || [];
  }

  async generateText(request: TextGenerationRequest): Promise<TextGenerationResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    const model = request.model || AIConfigManager.getDefaultModel('openai', 'text');
    const url = `${this.baseUrl}/chat/completions`;

    // 构建消息列表
    const messages: Array<{ role: string; content: string }> = [];

    if (request.systemInstruction) {
      messages.push({
        role: 'system',
        content: request.systemInstruction,
      });
    }

    if (request.messages && request.messages.length > 0) {
      messages.push(...request.messages.map(msg => ({
        role: msg.role,
        content: msg.content,
      })));
    } else {
      messages.push({
        role: 'user',
        content: request.prompt,
      });
    }

    const requestBody = {
      model,
      messages,
      temperature: request.temperature ?? 0.7,
      max_tokens: request.maxTokens ?? 2048,
    };

    try {
      const response = await this.makeRequest<any>(url, {
        method: 'POST',
        body: JSON.stringify(requestBody),
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
        },
      });

      const content = response.choices?.[0]?.message?.content || '';
      const usage = response.usage ? {
        inputTokens: response.usage.prompt_tokens || 0,
        outputTokens: response.usage.completion_tokens || 0,
        totalTokens: response.usage.total_tokens || 0,
      } : undefined;

      return {
        content,
        provider: 'openai',
        model,
        usage,
        finishReason: response.choices?.[0]?.finish_reason,
      };
    } catch (error) {
      throw new Error(`OpenAI text generation failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async generateTextStream(
    request: TextGenerationRequest,
    onChunk: (chunk: TextGenerationChunk) => void
  ): Promise<void> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    const model = request.model || AIConfigManager.getDefaultModel('openai', 'text');
    
    // 构建初始消息历史
    const messages: Array<any> = [];
    if (request.systemInstruction) {
      messages.push({
        role: 'system',
        content: request.systemInstruction,
      });
    }
    if (request.messages && request.messages.length > 0) {
      messages.push(...request.messages.map(msg => ({
        role: msg.role,
        content: msg.content,
      })));
    } else {
      messages.push({
        role: 'user',
        content: request.prompt,
      });
    }

    // 递归处理流式响应，支持多次 function call
    await this.processStreamWithFunctionCalls(
      model,
      messages,
      request,
      onChunk,
      0, // 递归深度
      10 // 最大递归深度（防止无限循环）
    );
  }

  /**
   * 处理流式响应，支持 Function Calling 的递归调用
   */
  private async processStreamWithFunctionCalls(
    model: string,
    messages: Array<any>,
    request: TextGenerationRequest,
    onChunk: (chunk: TextGenerationChunk) => void,
    depth: number,
    maxDepth: number
  ): Promise<void> {
    if (depth >= maxDepth) {
      console.warn('[OpenAIAdapter] 达到最大递归深度，停止处理 function calls');
      return;
    }

    const url = `${this.baseUrl}/chat/completions`;

    const requestBody: any = {
      model,
      messages,
      temperature: request.temperature ?? 0.7,
      max_tokens: request.maxTokens ?? 2048,
      stream: true,
    };

    // 添加 Function Calling 支持
    if (request.functionDefinitions && request.functionDefinitions.length > 0) {
      requestBody.tools = request.functionDefinitions.map(fn => ({
        type: 'function',
        function: {
          name: fn.name,
          description: fn.description,
          parameters: fn.parameters,
        },
      }));
    }

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${this.apiKey}`,
        },
        body: JSON.stringify(requestBody),
      });

      if (!response.ok) {
        throw new Error(`HTTP ${response.status}: ${await response.text()}`);
      }

      const reader = response.body?.getReader();
      const decoder = new TextDecoder();

      if (!reader) {
        throw new Error('Failed to get response reader');
      }

      let buffer = '';
      let functionCallBuffer: any = null;
      let assistantMessage: any = { role: 'assistant', content: '' };

      while (true) {
        const { done, value } = await reader.read();
        if (done) break;

        buffer += decoder.decode(value, { stream: true });
        const lines = buffer.split('\n');
        buffer = lines.pop() || '';

        for (const line of lines) {
          if (line.trim() === '' || !line.startsWith('data: ')) continue;
          
          const data = line.slice(6);
          if (data === '[DONE]') {
            // 检查是否有 function call 需要处理
            if (functionCallBuffer && request.onFunctionCall) {
              // 将 assistant message 添加到历史
              messages.push(assistantMessage);
              
              // 执行 function call
              try {
                const functionResult = await request.onFunctionCall(functionCallBuffer);
                
                // 将 function call 结果添加到消息历史
                messages.push({
                  role: 'function',
                  name: functionCallBuffer.name,
                  content: JSON.stringify(functionResult),
                });

                // 递归调用以继续流式响应
                await this.processStreamWithFunctionCalls(
                  model,
                  messages,
                  request,
                  onChunk,
                  depth + 1,
                  maxDepth
                );
                return;
              } catch (error) {
                console.error('[OpenAIAdapter] Function call execution failed:', error);
                // 继续处理，不中断
              }
            } else {
              onChunk({ content: '', done: true });
            }
            return;
          }

          try {
            const json = JSON.parse(data);
            const delta = json.choices?.[0]?.delta;
            const content = delta?.content || '';
            const functionCall = delta?.function_call;

            // 处理 Function Call
            if (functionCall) {
              if (functionCall.name) {
                functionCallBuffer = {
                  name: functionCall.name,
                  arguments: functionCall.arguments || '',
                };
                assistantMessage.function_call = {
                  name: functionCall.name,
                  arguments: functionCall.arguments || '',
                };
              } else if (functionCall.arguments && functionCallBuffer) {
                functionCallBuffer.arguments += functionCall.arguments;
                if (assistantMessage.function_call) {
                  assistantMessage.function_call.arguments += functionCall.arguments;
                }
              }
            }

            // 处理文本内容
            if (content) {
              assistantMessage.content += content;
              onChunk({
                content,
                done: false,
              });
            }

            // 检查是否完成
            const finishReason = json.choices?.[0]?.finish_reason;
            if (finishReason) {
              // 如果是 function_call，执行函数并递归
              if (finishReason === 'function_call' && functionCallBuffer && request.onFunctionCall) {
                // 将 assistant message 添加到历史
                messages.push(assistantMessage);
                
                // 执行 function call
                try {
                  const functionResult = await request.onFunctionCall(functionCallBuffer);
                  
                  // 将 function call 结果添加到消息历史
                  messages.push({
                    role: 'function',
                    name: functionCallBuffer.name,
                    content: JSON.stringify(functionResult),
                  });

                  // 递归调用以继续流式响应
                  await this.processStreamWithFunctionCalls(
                    model,
                    messages,
                    request,
                    onChunk,
                    depth + 1,
                    maxDepth
                  );
                  return;
                } catch (error) {
                  console.error('[OpenAIAdapter] Function call execution failed:', error);
                  // 继续处理，不中断
                }
              }

              const usage = json.usage ? {
                inputTokens: json.usage.prompt_tokens || 0,
                outputTokens: json.usage.completion_tokens || 0,
                totalTokens: json.usage.total_tokens || 0,
              } : undefined;

              onChunk({
                content: '',
                done: true,
                usage,
              });
            }
          } catch (e) {
            // 忽略解析错误
          }
        }
      }
    } catch (error) {
      throw new Error(`OpenAI stream generation failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async generateImage(request: ImageGenerationRequest): Promise<ImageGenerationResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['prompt']);

    const model = request.model || AIConfigManager.getDefaultModel('openai', 'image');
    const url = `${this.baseUrl}/images/generations`;

    const requestBody = {
      model,
      prompt: request.prompt,
      n: request.numberOfImages || 1,
      size: `${request.width || 1024}x${request.height || 1024}`,
      quality: 'standard',
    };

    try {
      const response = await this.makeRequest<any>(url, {
        method: 'POST',
        body: JSON.stringify(requestBody),
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
        },
      });

      const images = (response.data || []).map((img: any) => ({
        url: img.url,
        base64: img.b64_json,
      }));

      return {
        images,
        provider: 'openai',
        model,
        usage: {
          imagesGenerated: images.length,
        },
      };
    } catch (error) {
      throw new Error(`OpenAI image generation failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async textToSpeech(request: TextToSpeechRequest): Promise<AudioResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['text']);

    const model = request.model || AIConfigManager.getDefaultModel('openai', 'audio');
    const url = `${this.baseUrl}/audio/speech`;

    const requestBody = {
      model,
      input: request.text,
      voice: request.voice || 'alloy',
      speed: request.speed || 1.0,
    };

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          'Authorization': `Bearer ${this.apiKey}`,
        },
        body: JSON.stringify(requestBody),
      });

      if (!response.ok) {
        throw new Error(`HTTP ${response.status}: ${await response.text()}`);
      }

      const audioBlob = await response.blob();
      const audioUrl = URL.createObjectURL(audioBlob);

      return {
        audioUrl,
        provider: 'openai',
        model,
      };
    } catch (error) {
      throw new Error(`OpenAI TTS failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async speechToText(request: SpeechToTextRequest): Promise<AudioResponse> {
    this.ensureConfigured();
    this.validateRequest(request, ['audioFile']);

    const model = request.model || 'whisper-1';
    const url = `${this.baseUrl}/audio/transcriptions`;

    const formData = new FormData();
    formData.append('file', request.audioFile);
    formData.append('model', model);
    if (request.language) {
      formData.append('language', request.language);
    }

    try {
      const response = await fetch(url, {
        method: 'POST',
        headers: {
          'Authorization': `Bearer ${this.apiKey}`,
        },
        body: formData,
      });

      if (!response.ok) {
        throw new Error(`HTTP ${response.status}: ${await response.text()}`);
      }

      const data = await response.json();

      return {
        text: data.text,
        provider: 'openai',
        model,
        confidence: 1.0, // OpenAI不返回置信度
      };
    } catch (error) {
      throw new Error(`OpenAI STT failed: ${error instanceof Error ? error.message : String(error)}`);
    }
  }

  async generateVideo(request: VideoGenerationRequest): Promise<VideoGenerationResponse> {
    throw new Error('OpenAI video generation not supported');
  }
}

File: backend/src/main/java/com/heartsphere/aiagent/adapter/OpenAIAdapter.java
package com.heartsphere.aiagent.adapter;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.heartsphere.aiagent.dto.request.*;
import com.heartsphere.aiagent.dto.response.*;
import com.heartsphere.aiagent.exception.AIServiceException;
import com.heartsphere.aiagent.util.StreamResponseHandler;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.*;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestTemplate;

import java.util.*;
import java.util.Base64;

/**
 * OpenAI 适配器实现
 * 通过 HTTP 调用 OpenAI API
 * 
 * @author HeartSphere
 * @version 1.0
 */
@Slf4j
@Component
@RequiredArgsConstructor
public class OpenAIAdapter implements ModelAdapter {
    
    @Value("${spring.ai.openai.api-key:}")
    private String defaultApiKey;
    
    @Value("${spring.ai.openai.base-url:https://api.openai.com/v1}")
    private String baseUrl;
    
    private final RestTemplate restTemplate;
    private final ObjectMapper objectMapper;
    
    // 支持的文本模型
    private static final List<String> TEXT_MODELS = Arrays.asList(
        "gpt-4", "gpt-4-turbo", "gpt-3.5-turbo", "o1-preview", "o1-mini"
    );
    
    // 支持的图片模型
    private static final List<String> IMAGE_MODELS = Arrays.asList(
        "dall-e-3", "dall-e-2"
    );
    
    @Override
    public String getProviderType() {
        return "openai";
    }
    
    @Override
    public boolean supportsTextGeneration() {
        return true;
    }
    
    @Override
    public boolean supportsImageGeneration() {
        return true;
    }
    
    @Override
    public boolean supportsTextToSpeech() {
        return true;
    }
    
    @Override
    public boolean supportsSpeechToText() {
        return true;
    }
    
    @Override
    public boolean supportsVideoGeneration() {
        return false; // OpenAI 暂不支持视频生成
    }
    
    @Override
    public TextGenerationResponse generateText(TextGenerationRequest request) {
        try {
            log.debug("OpenAI文本生成请求: model={}, prompt={}", request.getModel(), request.getPrompt());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("OpenAI API key 未配置");
            }
            
            String url = baseUrl + "/chat/completions";
            
            // 构建请求体
            Map<String, Object> requestBody = buildOpenAIRequest(request);
            
            // 发送请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            headers.setBearerAuth(apiKey);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                throw new AIServiceException("OpenAI API 调用失败: " + response.getStatusCode());
            }
            
            // 解析响应
            return parseOpenAIResponse(response.getBody(), request);
            
        } catch (Exception e) {
            log.error("OpenAI文本生成失败", e);
            throw new AIServiceException("OpenAI文本生成失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public void generateTextStream(TextGenerationRequest request, 
                                   StreamResponseHandler<TextGenerationResponse> handler) {
        try {
            log.debug("OpenAI流式文本生成请求: model={}", request.getModel());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                handler.handle(null, true);
                throw new AIServiceException("OpenAI API key 未配置");
            }
            
            String url = baseUrl + "/chat/completions";
            
            // 构建请求体（设置 stream=true）
            Map<String, Object> requestBody = buildOpenAIRequest(request);
            requestBody.put("stream", true);
            
            // 发送流式请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            headers.setBearerAuth(apiKey);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            // 注意：这里简化实现，实际应该使用 SSE 或 WebClient 处理流式响应
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                handler.handle(null, true);
                throw new AIServiceException("OpenAI API 调用失败");
            }
            
            // 解析流式响应（简化实现）
            parseOpenAIStreamResponse(response.getBody(), request, handler);
            
        } catch (Exception e) {
            log.error("OpenAI流式文本生成失败", e);
            handler.handle(null, true);
        }
    }
    
    @Override
    public ImageGenerationResponse generateImage(ImageGenerationRequest request) {
        try {
            log.debug("OpenAI图片生成请求: model={}, prompt={}", request.getModel(), request.getPrompt());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("OpenAI API key 未配置");
            }
            
            String url = baseUrl + "/images/generations";
            
            // 构建请求体
            Map<String, Object> requestBody = new HashMap<>();
            requestBody.put("prompt", request.getPrompt());
            requestBody.put("model", request.getModel() != null ? request.getModel() : "dall-e-3");
            requestBody.put("n", 1); // 默认生成1张图片
            // 默认尺寸
            requestBody.put("size", "1024x1024");
            
            // 发送请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            headers.setBearerAuth(apiKey);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            ResponseEntity<JsonNode> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, JsonNode.class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                throw new AIServiceException("OpenAI 图片生成 API 调用失败");
            }
            
            // 解析响应
            return parseImageResponse(response.getBody(), request);
            
        } catch (Exception e) {
            log.error("OpenAI图片生成失败", e);
            throw new AIServiceException("OpenAI图片生成失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public AudioResponse textToSpeech(AudioRequest request) {
        try {
            log.debug("OpenAI文本转语音请求: text={}", request.getText());
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("OpenAI API key 未配置");
            }
            
            String url = baseUrl + "/audio/speech";
            
            // 构建请求体
            Map<String, Object> requestBody = new HashMap<>();
            requestBody.put("model", request.getModel() != null ? request.getModel() : "tts-1");
            requestBody.put("input", request.getText());
            requestBody.put("voice", request.getVoice() != null ? request.getVoice() : "alloy");
            
            // 发送请求
            HttpHeaders headers = new HttpHeaders();
            headers.setContentType(MediaType.APPLICATION_JSON);
            headers.setBearerAuth(apiKey);
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            ResponseEntity<byte[]> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, byte[].class
            );
            
            if (!response.getStatusCode().is2xxSuccessful() || response.getBody() == null) {
                throw new AIServiceException("OpenAI 文本转语音 API 调用失败");
            }
            
            AudioResponse result = new AudioResponse();
            result.setAudioBase64(Base64.getEncoder().encodeToString(response.getBody()));
            result.setProvider(getProviderType());
            result.setModel(request.getModel() != null ? request.getModel() : "tts-1");
            
            return result;
            
        } catch (Exception e) {
            log.error("OpenAI文本转语音失败", e);
            throw new AIServiceException("OpenAI文本转语音失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public AudioResponse speechToText(AudioRequest request) {
        try {
            log.debug("OpenAI语音转文本请求");
            
            // 获取 API key
            String apiKey = getApiKey(request);
            if (apiKey == null || apiKey.isEmpty()) {
                throw new AIServiceException("OpenAI API key 未配置");
            }
            
            // 简化实现：直接返回错误提示
            // 注意：OpenAI 语音转文本需要 multipart/form-data 请求
            throw new AIServiceException("OpenAI 语音转文本需要 multipart 请求，暂未实现");
            
        } catch (Exception e) {
            log.error("OpenAI语音转文本失败", e);
            throw new AIServiceException("OpenAI语音转文本失败: " + e.getMessage(), e);
        }
    }
    
    @Override
    public VideoGenerationResponse generateVideo(VideoGenerationRequest request) {
        throw new AIServiceException("OpenAI 暂不支持视频生成");
    }
    
    @Override
    public List<String> getSupportedModels(String capability) {
        switch (capability.toLowerCase()) {
            case "text":
                return TEXT_MODELS;
            case "image":
                return IMAGE_MODELS;
            case "audio":
                return Arrays.asList("tts-1", "tts-1-hd", "whisper-1");
            default:
                return new ArrayList<>();
        }
    }
    
    /**
     * 获取 API key
     */
    private String getApiKey(TextGenerationRequest request) {
        // TODO: 从管理后台配置中获取 API key
        return defaultApiKey;
    }
    
    private String getApiKey(ImageGenerationRequest request) {
        // TODO: 从管理后台配置中获取 API key
        return defaultApiKey;
    }
    
    private String getApiKey(AudioRequest request) {
        // TODO: 从管理后台配置中获取 API key
        return defaultApiKey;
    }
    
    /**
     * 构建 OpenAI API 请求体
     */
    private Map<String, Object> buildOpenAIRequest(TextGenerationRequest request) {
        Map<String, Object> requestBody = new HashMap<>();
        
        // 设置模型
        String model = request.getModel() != null ? request.getModel() : "gpt-3.5-turbo";
        requestBody.put("model", model);
        
        // 构建消息列表
        List<Map<String, String>> messages = new ArrayList<>();
        
        // 添加系统消息
        if (request.getSystemInstruction() != null && !request.getSystemInstruction().isEmpty()) {
            Map<String, String> systemMsg = new HashMap<>();
            systemMsg.put("role", "system");
            systemMsg.put("content", request.getSystemInstruction());
            messages.add(systemMsg);
        }
        
        // 添加历史消息
        if (request.getMessages() != null && !request.getMessages().isEmpty()) {
            for (TextGenerationRequest.Message msg : request.getMessages()) {
                Map<String, String> message = new HashMap<>();
                message.put("role", msg.getRole());
                message.put("content", msg.getContent());
                messages.add(message);
            }
        }
        
        // 添加当前提示词
        if (request.getPrompt() != null && !request.getPrompt().isEmpty()) {
            Map<String, String> userMsg = new HashMap<>();
            userMsg.put("role", "user");
            userMsg.put("content", request.getPrompt());
            messages.add(userMsg);
        }
        
        requestBody.put("messages", messages);
        
        // 添加生成参数
        if (request.getTemperature() != null) {
            requestBody.put("temperature", request.getTemperature());
        } else {
            requestBody.put("temperature", 0.7);
        }
        if (request.getMaxTokens() != null) {
            requestBody.put("max_tokens", request.getMaxTokens());
        }
        
        return requestBody;
    }
    
    /**
     * 解析 OpenAI API 响应
     */
    private TextGenerationResponse parseOpenAIResponse(JsonNode response, TextGenerationRequest request) {
        TextGenerationResponse result = new TextGenerationResponse();
        result.setProvider(getProviderType());
        result.setModel(request.getModel() != null ? request.getModel() : "gpt-3.5-turbo");
        
        try {
            // 提取内容
            JsonNode choices = response.get("choices");
            if (choices != null && choices.isArray() && choices.size() > 0) {
                JsonNode choice = choices.get(0);
                JsonNode message = choice.get("message");
                if (message != null && message.has("content")) {
                    result.setContent(message.get("content").asText());
                }
                
                // 提取完成原因
                if (choice.has("finish_reason")) {
                    result.setFinishReason(choice.get("finish_reason").asText());
                }
            }
            
            // 提取 Token 使用量
            JsonNode usage = response.get("usage");
            if (usage != null) {
                TextGenerationResponse.TokenUsage tokenUsage = new TextGenerationResponse.TokenUsage();
                if (usage.has("prompt_tokens")) {
                    tokenUsage.setInputTokens(usage.get("prompt_tokens").asInt());
                }
                if (usage.has("completion_tokens")) {
                    tokenUsage.setOutputTokens(usage.get("completion_tokens").asInt());
                }
                if (usage.has("total_tokens")) {
                    tokenUsage.setTotalTokens(usage.get("total_tokens").asInt());
                }
                result.setUsage(tokenUsage);
            }
            
        } catch (Exception e) {
            log.warn("解析 OpenAI 响应失败", e);
            result.setContent("");
        }
        
        return result;
    }
    
    /**
     * 解析 OpenAI 流式响应
     */
    private void parseOpenAIStreamResponse(JsonNode response, TextGenerationRequest request,
                                           StreamResponseHandler<TextGenerationResponse> handler) {
        try {
            // 简化实现：解析流式响应
            // 实际应该处理多个 SSE chunk
            JsonNode choices = response.get("choices");
            if (choices != null && choices.isArray()) {
                for (JsonNode choice : choices) {
                    JsonNode delta = choice.get("delta");
                    if (delta != null && delta.has("content")) {
                        String text = delta.get("content").asText();
                        TextGenerationResponse chunk = new TextGenerationResponse();
                        chunk.setProvider(getProviderType());
                        chunk.setModel(request.getModel() != null ? request.getModel() : "gpt-3.5-turbo");
                        chunk.setContent(text);
                        handler.handle(chunk, false);
                    }
                }
            }
            
            // 完成
            TextGenerationResponse finalResponse = new TextGenerationResponse();
            finalResponse.setProvider(getProviderType());
            finalResponse.setModel(request.getModel() != null ? request.getModel() : "gpt-3.5-turbo");
            handler.handle(finalResponse, true);
            
        } catch (Exception e) {
            log.error("解析 OpenAI 流式响应失败", e);
            handler.handle(null, true);
        }
    }
    
    /**
     * 解析图片生成响应
     */
    private ImageGenerationResponse parseImageResponse(JsonNode response, ImageGenerationRequest request) {
        ImageGenerationResponse result = new ImageGenerationResponse();
        result.setProvider(getProviderType());
        result.setModel(request.getModel() != null ? request.getModel() : "dall-e-3");
        
        try {
            List<ImageGenerationResponse.Image> images = new ArrayList<>();
            JsonNode data = response.get("data");
            if (data != null && data.isArray()) {
                for (JsonNode item : data) {
                    ImageGenerationResponse.Image img = new ImageGenerationResponse.Image();
                    if (item.has("url")) {
                        img.setUrl(item.get("url").asText());
                    }
                    if (item.has("b64_json")) {
                        img.setBase64(item.get("b64_json").asText());
                    }
                    images.add(img);
                }
            }
            result.setImages(images);
            
            ImageGenerationResponse.Usage usage = new ImageGenerationResponse.Usage();
            usage.setImagesGenerated(images.size());
            result.setUsage(usage);
            
        } catch (Exception e) {
            log.warn("解析 OpenAI 图片响应失败", e);
            result.setImages(new ArrayList<>());
        }
        
        return result;
    }
}

